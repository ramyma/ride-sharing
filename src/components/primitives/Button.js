import PropTypes from 'prop-types';
import styled, { css } from '@xstyled/emotion';
import Box from './Box';

const Button = styled(Box)`
  cursor: pointer;
  user-select: none;
  font-weight: bold;

  transition: 280ms all ease;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.4;
  }

  ${props =>
    !props.disabled &&
    css`
      &:hover {
        background-color: ${props.inverse
          ? props.backgroundColor
          : props.color};
        color: ${props.inverse ? props.color : props.backgroundColor};
      }
    `}

  ${props =>
    props.inverse &&
    css`
      background-color: ${props.color};
      color: ${props.backgroundColor};
    `}
`;

Button.defaultProps = {
  as: 'button',
  backgroundColor: 'background',
  border: '1px solid',
  borderRadius: 4,
  color: 'inverseBackground',
  height: 'fit-content',
  borderColor: 'inverseBackground',
  p: 2,
  minWidth: 100
};

Button.propTypes = {
  disabled: PropTypes.bool,
  inverse: PropTypes.bool
};

export default Button;
