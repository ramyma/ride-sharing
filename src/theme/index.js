import { opacify } from 'polished';

const theme = {
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  },
  defaultColorModeName: 'light',
  colors: {
    background: '#fff',
    grey: '#585858',
    inverseBackground: '#1c1c1c',
    shadow: opacify(0.19, '#000'),
    textDark: '#353535',
    text: '#353535',
    primary: '#ee424a',
    success: '#4CAF50',
    danger: '#f44336',
    warning: '#ffa000',
    modes: {
      dark: {
        background: '#1c1c1c',
        grey: '#9e9e9e',
        inverseBackground: '#fff',
        shadow: opacify(0.19, '#fff'),
        text: '#fff'
      }
    }
  },
  fontSizes: [0, 12, 14, 16, 20, 24, 32, 48, 64, 72],
  sizes: [0, 2, 4, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32],
  shadows: {
    grey: 'rgba(0, 0, 0, 0.19) 0 0 10px 0px',
    greyInset: `inset rgba(0, 0, 0, 0.19) 0 0 10px 0px`,
    pink: '#e42974 0 0 3px',
    text: '0 0 rgba(0, 0, 0, 0.19)'
  },
  space: [0, 4, 8, 16, 24, 48, 96, 144, 192, 240]
};

export default theme;
