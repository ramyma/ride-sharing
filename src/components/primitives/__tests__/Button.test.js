import React from 'react';
import * as emotion from '@xstyled/emotion';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@xstyled/emotion';
import { createSerializer } from 'jest-emotion';
import Button from '../Button';
import theme from '../../../theme';

expect.addSnapshotSerializer(createSerializer(emotion));

it('Button renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Button>Test</Button>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button renders with inverse prop correctly', () => {
  const tree = renderer.create(<Button inverse>Test</Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Button renders with disabled prop correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Button disabled>Test</Button>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
