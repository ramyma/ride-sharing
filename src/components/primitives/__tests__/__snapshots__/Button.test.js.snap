// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.emotion-0 {
  background-color: #fff;
  border: 1px solid;
  border-radius: 4px;
  border-color: #1c1c1c;
  color: #1c1c1c;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  min-width: 100px;
  padding: 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: bold;
  -webkit-transition: 280ms all ease;
  transition: 280ms all ease;
}

.emotion-0:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

.emotion-0:hover {
  background-color: #1c1c1c;
  color: #fff;
}

<button
  className="emotion-0"
  color="inverseBackground"
  height="fit-content"
>
  Test
</button>
`;

exports[`Button renders with disabled prop correctly 1`] = `
.emotion-0 {
  background-color: #fff;
  border: 1px solid;
  border-radius: 4px;
  border-color: #1c1c1c;
  color: #1c1c1c;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  min-width: 100px;
  padding: 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: bold;
  -webkit-transition: 280ms all ease;
  transition: 280ms all ease;
}

.emotion-0:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

<button
  className="emotion-0"
  color="inverseBackground"
  disabled={true}
  height="fit-content"
>
  Test
</button>
`;

exports[`Button renders with inverse prop correctly 1`] = `
.emotion-0 {
  background-color: background;
  border: 1px solid;
  border-radius: 4px;
  border-color: inverseBackground;
  color: inverseBackground;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  min-width: 100px;
  padding: 8px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: bold;
  -webkit-transition: 280ms all ease;
  transition: 280ms all ease;
  background-color: inverseBackground;
  color: background;
}

.emotion-0:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

.emotion-0:hover {
  background-color: background;
  color: inverseBackground;
}

<button
  className="emotion-0"
  color="inverseBackground"
  height="fit-content"
>
  Test
</button>
`;
